src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
+          // Don't know how to place a URI object in JsonNode, keeping the text value as of now
+        case s: TextNode => (s, Array[String](), csvwPropertyType)
+        case _ => (new TextNode(""), Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def textDirectionProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode if Array[String]("ltr", "rtl", "inherit").contains(s.asText()) => {
+          (value, Array[String](), PropertyType.Inherited)
+        }
+        case _ => (new TextNode(PropertyType.Inherited.toString), Array[String](PropertyChecker.invalidValueWarning), PropertyType.Inherited)
+      }
+    }
+  }
+
+  def naturalLanguageProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      var warnings = Array[String]()
+      value match {
+        case s:TextNode => {
+          val returnObject = JsonNodeFactory.instance.objectNode()
+          val arrayNode = JsonNodeFactory.instance.arrayNode()
+          arrayNode.add(s.asText())
+          returnObject.set("lang", arrayNode)
+          (returnObject, Array[String](), csvwPropertyType)
+        }
+        case a:ArrayNode => {
+          var validTitles = Array[String]()
+          val arrayNodeIterator = a.elements()
+          while(arrayNodeIterator.hasNext()) {
+            val element = arrayNodeIterator.next()
+            element match {
+              case s:TextNode => validTitles = validTitles :+ s.asText()
+              case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+            }
+          }
+          val returnObject = JsonNodeFactory.instance.objectNode()
+          val arrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(validTitles)
+          returnObject.set("lang", arrayNode)
+          (returnObject, warnings, csvwPropertyType)
+        }
+        case o:ObjectNode => {
+          val valueCopy = o.deepCopy()
+          val fieldsAndValues = Array.from(valueCopy.fields.asScala)
+          for(fieldsAndValue <- fieldsAndValues) {
+            val elementKey = fieldsAndValue.getKey
+            val matcher = PropertyChecker.Bcp47LanguagetagRegExp.pattern.matcher(elementKey)
+            if(matcher.matches()) {
+              var validTitles = Array[String]()
+              fieldsAndValue.getValue match {
+                case s:TextNode => validTitles = validTitles :+ s.asText()
+                case a:ArrayNode => {
+                  val titles = Array.from(a.elements().asScala)
+                  for(title <- titles) {
+                    title match {
+                      case s:TextNode => validTitles = validTitles :+ s.asText()
+                      case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
```

Comment:
Again, I'd be tempted to let them know what the invalid value is. Given it's a JSON node, it should be very easy to serialise to JSON (and truncate if it's too long for sensible CLI output).

Reply:
Given you'll be doing this in two places, a function to do it would be a good plan.

Have a think about whether there any other locations which could benefit from this.

Reply:
Done